connection.js

```diff
@@ -375,82 +381,61 @@ class Connection {
-        }
-        if (!config.allowed_timecontrols_ranked[notification.time_control.time_control] && notification.ranked && config.timecontrolsranked) { 
-            return genericAllowedFamiliesReject("timecontrolsranked", notification.time_control.time_control);
-        }
-        if (!config.allowed_timecontrols_unranked[notification.time_control.time_control] && !notification.ranked && config.timecontrolsunranked) { 
-            return genericAllowedFamiliesReject("timecontrolsunranked", notification.time_control.time_control);
-        }
+        const resultTimecontrols = getAllowedFamilyRejectResult("timecontrols", "Time control", notification.time_control.time_control, notification.ranked);
+        if (resultTimecontrols) return resultTimecontrols;
 
         return { reject: false }; // OK !
 
     }
-    // Check challenge settings are allowed
+    // Check challenge minMax are allowed
     //
-    checkChallengeSettings(notification) {
+    checkChallengeMinMax(notification) {
 
-        const handicapSettings =  { notif: notification.handicap, isFakeHandicap: config.fakerank || false };
-        if (notification.handicap === -1 && config.fakerank) {
-            // TODO: modify or remove fakerank code whenever server sends us automatic handicap
-            //       notification.handicap different from -1.
-            // adding a .floor: 5.9k (6k) vs 6.1k (7k) is 0.2 rank difference,
-            // but it is still a 6k vs 7k = 1 rank difference = 1 automatic handicap stone
-            handicapSettings.notif = Math.abs(Math.floor(notification.user.ranking) - Math.floor(config.fakerank));
-        }
-        const resultHandicap = minMaxHandicapRankRejectResult("handicap", handicapSettings.notif, handicapSettings.isFakeHandicap, notification.ranked);
+        // minMax rank is checked earlier in checkChallengeMandatory, the rest of the minMax are here.
+
+        const resultHandicap = getMinMaxHandicapRejectResult(notification.handicap, notification.user.ranking, notification.ranked);
         if (resultHandicap) return resultHandicap;
-        const resultMaintime = UHMAEATRejectResult("maintime", notification.time_control, notification.ranked);
-        if (resultMaintime) return resultMaintime;
-        const resultPeriods = minMaxPeriodsRejectResult("periods", notification.time_control.periods, notification.time_control.speed, notification.ranked);
-        if (resultPeriods) return resultPeriods;
-        const resultPeriodtime = UHMAEATRejectResult("periodtime", notification.time_control, notification.ranked);
-        if (resultPeriodtime) return resultPeriodtime;
+
+        // time control "none" has no maintime, no periods number, no periodtime, no need to check reject.
+        // also this avoids an undefined error in getTimecontrolObjsMainPeriodTime when getting timesObj.(maintime|periodtime).none
+        if (notification.time_control.time_control !== "none") {
```

Comment:
ah but this could create problems, like if we add another minmax in the future or move handicap to the bottom, then the `return { reject: false };` statement would early exit `checkChallengeMinMax` before doing possible future checks we add in the future